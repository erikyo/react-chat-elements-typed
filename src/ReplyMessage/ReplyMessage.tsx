import type React from "react";
import type { FC } from "react";
import "./ReplyMessage.css";

import classNames from "classnames";
import type { IReplyMessageProps } from "../types";

const ReplyMessage: FC<IReplyMessageProps> = (props) => {
	return (
		<div
			className={classNames("rce-mbox-reply", {
				"rce-mbox-reply-border": !!props.titleColor,
			})}
			style={{ ...(props.titleColor && { borderColor: props.titleColor }) }}
			onClick={props.onClick}
			onKeyDown={console.log}
		>
			<div className="rce-mbox-reply-left">
				<div
					style={{ ...(props.titleColor && { color: props.titleColor }) }}
					className="rce-mbox-reply-owner"
				>
					{props.title}
				</div>
				<div className="rce-mbox-reply-message">{props.text || "..."}</div>
			</div>
			{props.photoURL && (
				<div className="rce-mbox-reply-right">
					<img src={props.photoURL} alt="" />
				</div>
			)}
		</div>
	);
};

export default ReplyMessage;
