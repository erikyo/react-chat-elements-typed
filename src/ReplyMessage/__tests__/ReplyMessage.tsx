import { describe, expect, it } from "vitest";
import React from "react";
import ReplyMessage from "../ReplyMessage";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<ReplyMessage />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
