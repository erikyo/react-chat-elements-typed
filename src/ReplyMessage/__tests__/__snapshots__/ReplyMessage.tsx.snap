// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-mbox-reply"
  >
    <div
      class="rce-mbox-reply-left"
    >
      <div
        class="rce-mbox-reply-owner"
      />
      <div
        class="rce-mbox-reply-message"
      >
        ...
      </div>
    </div>
  </div>
</div>
`;
